src/util/defaults.ml

```diff
@@ -181,6 +181,24 @@ let _ = ()
 let _ = ()
       ; reg Annotation "annotation.int.enabled"   "false" "Enable manual annotation of functions with desired precision, i.e. the activated IntDomains."
       ; reg Annotation "annotation.int.privglobs" "true"  "Enables handling of privatized globals, by setting the precision to the heighest value, when annotation.int.enabled is true."
+      ; reg Annotation "annotation.goblint_context.base.no-non-ptr" "[]" ""
+      ; reg Annotation "annotation.goblint_context.base.non-ptr" "[]" ""
+      ; reg Annotation "annotation.goblint_context.base.no-int" "[]" ""
+      ; reg Annotation "annotation.goblint_context.base.int" "[]" ""
+      ; reg Annotation "annotation.goblint_context.base.no-interval" "[]" ""
+      ; reg Annotation "annotation.goblint_context.base.interval" "[]" ""
+      ; reg Annotation "annotation.goblint_context.apron.no-context" "[]" ""
+      ; reg Annotation "annotation.goblint_context.apron.context" "[]" ""
+      ; reg Annotation "annotation.goblint_context.no-widen" "[]" ""
+      ; reg Annotation "annotation.goblint_context.widen" "[]" ""
+      ; reg Annotation "annotation.goblint_precision.no-def_exc" "[]" ""
+      ; reg Annotation "annotation.goblint_precision.def_exc" "[]" ""
+      ; reg Annotation "annotation.goblint_precision.no-interval" "[]" ""
+      ; reg Annotation "annotation.goblint_precision.interval" "[]" ""
+      ; reg Annotation "annotation.goblint_precision.no-enums" "[]" ""
+      ; reg Annotation "annotation.goblint_precision.enums" "[]" ""
+      ; reg Annotation "annotation.goblint_precision.no-congruence" "[]" ""
+      ; reg Annotation "annotation.goblint_precision.congruence" "[]" ""
```

Comment:
This is all very verbose, any suggestions on how to fix this?

Reply:
Maybe use JSON objects for values instead?
```json
{
    "fun1": ["base.int", "widen"],
    "fun2": ["no-widen"]
}
```

Reply:
Yes, I also thought about that, it does indeed appear cleaner.

The downside seems to be that we either

- lose the ability to conveniently set this from the command line and
- lose the ability to reject malformed options such as `non_ptr` instead of `non-ptr`

or need custom code to support these things in such a setting as well.

Maybe for now we go with this more verbose form and rethink it as part of #476 ?

Reply:
> * lose the ability to conveniently set this from the command line and

If the default value is `{}`, then it's just another JSON object in our entire options object, so `--set annotation.goblint_context.fun1[+] base.int` should work just as well I think.



> * lose the ability to reject malformed options such as `non_ptr` instead of `non-ptr`

We already lose that when using annotations in the code: we don't fail for unknown annotations and that's how it should be anyway. The ability to do it for options would be just extra safety. With #476 it might become possible to actually specify that as well.

Reply:
> If the default value is {}, then it's just another JSON object in our entire options object, so --set annotation.goblint_context.fun1[+] base.int should work just as well I think.

No, it complains about `Unknown path`. One would have to modify it to just create any path that does not exist yet. I'm skeptical if we want that.

Reply:
Right. I guess I'll just have to refactor all this in #476 then, because there it would be completely safe to allow new subobjects to be created at any time. The schema would validate whether a certain object allows arbitrary keys or not. In fact the schema for my suggested way would be notably simpler and less repetitive than it is for this.